docs/versioned/admin/editing-configmaps.md

```diff
@@ -0,0 +1,155 @@
+
+### Store ConfigMaps as code
+
+Define ConfigMaps in YAML or JSON files and store them in a version control system (VCS) like Git. For example:
+
+    ```yml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: my-app-config
+      namespace: default
+    data:
+      _example | ...
+      app.properties: |
+        version=1.2.3
+        db.host=localhost
+        db.port=5432
+    ```
+
+- Commit these files to a Git repository to track changes over time.
+
+### Versioning ConfigMaps in Git
+
+- Use meaningful commit messages to describe changes to ConfigMap data (e.g., "Updated app.properties to version 1.2.4").
+- Tag commits in Git with version numbers (e.g., `git tag config-v1.2.3`) to mark specific ConfigMap versions.
+- Use branches for different environments (e.g., `dev`, `staging`, `prod`) or feature-specific ConfigMap changes.
+
+### Track Object Version in ConfigMap
+
+- Include a version field in the ConfigMap’s data to explicitly track the version of the application or configuration it represents.
+
+    ```yaml
+    data:
+    app.properties: |
+      version=1.2.3
+      # other settings
+    ```
+
+- Alternatively, use annotations in the ConfigMap’s metadata:
+
+    ```yaml
+    metadata:
+      name: my-app-config
+      annotations:
+        app-version: "1.2.3"
+    ```
+
+### Use GitOps for Deployment
+
+- Implement a GitOps workflow with tools like ArgoCD or Flux to synchronize ConfigMaps from Git to your Kubernetes cluster.
+- These tools detect changes in the Git repository and automatically apply them to the cluster, ensuring the deployed ConfigMap matches the versioned configuration in Git.
+
+### Immutable ConfigMaps with Versioned Names
+
+- Create immutable ConfigMaps by appending a version or hash to the ConfigMap name (e.g., `my-app-config-v1.2.3` or `my-app-config-abc123`).
+- Update the application’s Deployment or Pod to reference the new ConfigMap version when rolling out changes.
+
+    ```yaml
+    apiVersion: apps/v1
+    kind: Deployment
+    spec:
+      template:
+        spec:
+          containers:
+          - name: my-app
+            envFrom:
+            - configMapRef:
+                name: my-app-config-v1.2.3
+    ```
+
+- This approach ensures ConfigMaps are not modified in-place, preserving historical versions.
+
+### Track Changes in Kubernetes
+
+- Use Kubernetes resource versioning (e.g., `resourceVersion` in metadata) to track changes to the ConfigMap object itself within the cluster. However, this is managed by Kubernetes and not suitable for application-level versioning.
+- Enable audit logging in Kubernetes to track who modified ConfigMaps and when.
+
+### Use Helm for Versioned Configurations
+
+- If using Helm, store ConfigMaps in Helm chart templates or values files and version the chart itself (e.g., `Chart.yaml` with `version: 1.2.3`).
+- Helm’s release history can track which ConfigMap versions were deployed.
```

Comment:
Helm can be used this way, but neither Serving nor Eventing currently publishes a helm chart, so this is probably not helpful.